lib/google/gax/configuration.rb

```diff
@@ -112,15 +116,21 @@ def self.config? obj
       #     validation fails. Defaults to `true`.
       #
       def initialize show_warnings: true, &block
-        @show_warnings = show_warnings
         @values = {}
-        @defaults = {}
-        @validators = {}
+        @schema = Schema.new show_warnings: show_warnings
 
         # Can't call yield because of BasicObject
         block&.call self
       end
 
+      ##
+      # Derive a Configuration object. The subsequent object can set local values,
+      # but will not be able to change the structure.
+      #
+      def derive! &block
```

Comment:
I do not like this name. I really don't. But I can't think of anything better and so I went with it. Hopefully someone can suggest a far superior name for what this does.

Reply:
I don't mind it. OO terminology seems appropriate.